### PR TITLE
Simplify kube-scheduler's run() method

### LINKED ISSUE #68964: [scheduler cleanup phase 1]: Simplify kube-scheduler’s server.go `run`
**PLEASE COORDINATE WITH SIG-SCHEDULING BEFORE WORKING ON THIS ISSUE.** We want to avoid the situation where two people accidentally start work on the same fix.
Main tracking issue: https://github.com/kubernetes/kubernetes/issues/68951

In `cmd/kube-scheduler/app/server.go` the `NewSchedulerCommand()` function creates a `cobra.Command` object and populates the `Run` field with an anonymous function. At the end of that function, `Run()` is called, which contains the rest of our startup logic.

Rather than split this across two functions, we should put all the startup logic in one named function called `run` (not `Run`) so that constructing the `cobra.Command` object is much more plain.

/sig scheduling
/kind cleanup
/help
/good-first-issue

### PR DESCRIPTION
**Pull request by shubheksha**
**What this PR does / why we need it**:
This PR simplifies kube-scheduler's `run()` method by consolidating the logic into the already existing `run()` method.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #68964 
Addresses #68951

**Special notes for your reviewer**: N/A

**Release note**:

```release-note
NONE
```
/sig scheduling
/kind cleanup
/cc @misterikkit

### CODE REVIEW

File: cmd/kube-scheduler/app/server.go [hunk 1]
```diff
@@ -123,7 +95,38 @@ through the API as necessary.`,
 }
 
 // Run runs the Scheduler.
```
Comment (misterikkit) [inline cmd/kube-scheduler/app/server.go]: Please change code comment to lowercase also.

**Review by misterikkit — APPROVED** [review]
/lgtm
Minor nit on the code comment.

File: cmd/kube-scheduler/app/server.go (hunk 1 above)
Comment (shubheksha) [reply cmd/kube-scheduler/app/server.go]: Done! 👍

**Review by misterikkit — APPROVED** [review]
/lgtm

**Pull request merged**